src/rest/RequestHandler.js

```diff
@@ -100,6 +128,9 @@ class RequestHandler {
         // Set the variables based on the route-specific rate limit
         limit = this.limit;
         timeout = this.reset + this.manager.client.options.restTimeOffset - Date.now();
+
+        await this.onRateLimit(request, limit, timeout, isGlobal); // eslint-disable-line no-await-in-loop
```

Comment:
Could you move these two calls after the event emitting? So, after that if check, run this?